prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -694,4 +707,24 @@ class FHIRRouter(
                 ?: emptyList()
             ).plus(receiver.mappedConditionFilter)
     }
+
+    private fun emitAzureEvent(
+        report: Report,
+        message: ReportPipelineMessage,
+        receiver: Receiver?,
+        bundle: Bundle,
+    ) {
+        val sender = reportService.getSenderName(message.reportId)
+        val conditions = bundle.getAllMappedConditions()
+
+        azureEventService.trackEvent(
+            ReportRouteEvent(
```

Comment:
If we used builder pattern, we could just call this `UniversalPipelineEvent` instead, have a field for "type" (route, convert, etc), and then just set the params that make sense for that particular case. We wouldn't have to update all the usages everywhere. What do you think about that? Any other advantages or disadvantages? Perhaps this is something we just wait to refactor if we find we need to.

Reply:
I think events should be different enough that making a single class with a builder might end up too large and confusing.

I also think the compiler errors when we update an event definition are beneficial! We don't want to add a new field and possibly miss a spot and then end up emitting bad events for months unknowingly.